src/integrationTest/java/uk/gov/hmcts/reform/ccd/TestClientRetry.java

```diff
@@ -0,0 +1,176 @@
+import static uk.gov.hmcts.reform.ccd.util.RestConstants.DELETE_TASKS_PATH;
+import static uk.gov.hmcts.reform.ccd.util.RestConstants.LAU_SAVE_PATH;
+
+@SpringBootTest
+@ActiveProfiles("test")
+@ComponentScan({"uk.gov.hmcts.reform.ccd"})
+@TestPropertySource(properties = {
+    "remote.document.store.host=http://localhost:8080",
+    "remote.role.assignment.host=http://localhost:8080",
+    "remote.log.and.audit.host=http://localhost:8080",
+    "remote.tasks.host=http://localhost:8080",
+    "remote.hearing.host=http://localhost:8080",
+})
+class TestClientRetry {
+    protected static final WireMockServer WIREMOCK_SERVER = new WireMockServer(8080);
+
+    static {
+        if (!WIREMOCK_SERVER.isRunning()) {
+            WIREMOCK_SERVER.start();
+        }
+
+        WIREMOCK_SERVER.stubFor(post(urlPathMatching(DELETE_DOCUMENT_PATH))
+                                    .willReturn(aResponse()
+                                                    .withStatus(502)
+                                                    .withBody("Document Delete : Bad gateway")));
+
+        WIREMOCK_SERVER.stubFor(post(urlPathMatching(DELETE_ROLE_PATH))
+                                    .willReturn(aResponse()
+                                                    .withStatus(502)
+                                                    .withBody("Role Assignment Delete : Bad gateway")));
+
+        WIREMOCK_SERVER.stubFor(post(urlPathMatching(LAU_SAVE_PATH))
+                                    .willReturn(aResponse()
+                                                    .withStatus(502)
+                                                    .withBody("Lau Save : Bad gateway")));
+
+        WIREMOCK_SERVER.stubFor(post(urlPathMatching(DELETE_TASKS_PATH))
+                                    .willReturn(aResponse()
+                                                    .withStatus(502)
+                                                    .withBody("Task Delete : Bad gateway")));
+
+        WIREMOCK_SERVER.stubFor(delete(urlPathMatching(DELETE_HEARINGS_PATH))
+                                    .willReturn(aResponse()
+                                                    .withStatus(502)
+                                                    .withBody("Document Delete : Bad gateway")));
+
+    }
+
+    @Autowired
+    private DocumentClient documentClient;
+
+    @Autowired
+    private RoleAssignmentClient roleAssignmentClient;
+
+    @Autowired
+    private TasksClient tasksClient;
+
+    @Autowired
+    private LauClient lauClient;
+
+    @Autowired
+    private HearingClient hearingClient;
+
+    @Test
+    void testFeignDocumentClientRetry() throws InterruptedException {
+
+
+        DocumentsDeletePostRequest request = new DocumentsDeletePostRequest("12345");
+
+        try {
+            documentClient.deleteDocument("serviceAuthHeader", request);
+        } catch (Exception e) {
+            System.out.println("Exception: " + e.getMessage());
```

Comment:
Maybe we can do further verification here? Obviously if this isn't possible System.out.println is not making any good in the tests